spray-http/src/main/scala/spray/http/Uri.scala

```diff
@@ -158,10 +158,9 @@ sealed abstract case class Uri(scheme: String, authority: Authority, path: Path,
    */
   def renderWithoutFragment[R <: Rendering](r: R, charset: Charset): r.type = {
     if (isAbsolute) r ~~ scheme ~~ ':'
-    authority.render(r, scheme, charset)
+    authority.render(r, scheme, path.isEmpty || path.startsWithSlash, charset)
```

Comment:
Could we maybe simply pass the `path` to the render method and have the `path.isEmpty || path.startsWithSlash` condition be evaluated there?
This would have the benefit of keeping all the logic required for rendering inside of the `render` method, rather than moving some of it outside into argument construction.